cita-executor/core/src/state/account.rs

```diff
@@ -271,7 +272,7 @@ impl Account {
     /// Verify value proof of the trie's storage at `key`.
     pub fn verify_value_proof(&self, key: &H256, proof: &[Bytes]) -> Option<H256> {
         trie::triedb::verify_value_proof(key, self.storage_root, proof, ::rlp::decode)
-            .map(|v: U256| v.into())
+            .map(&Into::into as &Fn(U256) -> H256)
```

Comment:
I guess `&Into::into as &Fn(_) -> H256` is ok, too.

Reply:
I can't try now. will check it later.
But why not `&Into::into as &Fn(U256) -> _`?

Reply:
The input of `Into::into` is determined by `verify_value_proof(..)`.
So rust compiler can know this **implicitly**.

But rust compiler can't know the type of output.
So you have to tell rust compiler **explicitly**.

In fact, if the rust compiler check the function signature, I think it can know the type of output.
Maybe the output `Option<H256>` is **out of the scope** which it uses to determine the type.

Example:
```rust
fn f<T>(x: Vec<T>) -> Vec<T> { x }
fn g<T>(x: T) -> T { x }
fn main() {
    // failed
    let _ = (1..100).collect();
    // passed
    let _ = (1..100).collect::<Vec<_>>();
    // failed
    let t = (1..100).collect();
    let _ = g(t);
    // passed
    let t = (1..100).collect();
    let t = g(t);
    let _ = f(t);

}
```